src/net.h

```diff
@@ -614,12 +614,27 @@ class CNetMessage {
     int readData(const char *pch, unsigned int nBytes);
 };
 
+/** The TransportSerializer prepares messages for the network transport
+ */
+class TransportSerializer {
+public:
+    // prepare message for transport (insert header, etc.)
+    virtual void prepareForTransport(CSerializedNetMsg& msg, std::vector<unsigned char>& header) = 0;
+    virtual ~TransportSerializer() {}
```

Comment:
Do you expect serializer to keep state between call and so need to clean it with a destructor? (or just a good practice)

Reply:
Since we keep the instances in an std::unique_ptr, we need to have a virtual destructor.